src/python/dxpy/scripts/dx.py

```diff
@@ -3785,6 +3785,155 @@ def publish(args):
     except:
         err_exit()
 
+def archive(args):
+    def send_archive_request(target_project, request_input, request_func):
+        api_errors = [InvalidState, ResourceNotFound, PermissionDenied]
+        try:
+            res = request_func(target_project, request_input)
+        except Exception as e:            
+            eprint("Failed request: {}".format(request_input))
+            if type(e) in api_errors:
+                eprint("     API error: {}. {}".format(e.name, e.msg))
+            else: 
+                eprint("     Unexpected error: {}".format(format_exception(e)))
+            
+            err_exit("Failed request: {}. {}".format(request_input, format_exception(e)), code=3)
+        return res              
+
+    def get_valid_archival_input(args, target_files, target_folder, target_project):
+        request_input = {}
+        if target_files and target_folder:
+            err_exit('Expecting either a single folder or a list files for each API request', code=3)
+        elif target_files: 
+            request_input = {"files": target_files}
+        elif target_folder:
+            request_input = {"folder": target_folder, "recurse":args.recurse}
+        else:
+            err_exit("No input file/folder is found in project {}".format(target_project), 
+                            code=3)
+        return request_input
+
+    def get_archival_paths(args):
```

Comment:
For procedures like this, it might be nice to do some more detailed unit testing to confirm various corner cases and general expected behavior.   While I wouldn't make it a 'requirement', it would be much appreciated by myself and future us 😃

Reply:
More tests have been added :)

Reply:
Updates on getting project id: 
- If the __target project__ is not set, check the path and validate its project by calling "describe", set it as the target project
- if after processing there is neither target folder nor target files, reset the target project to NONE, so the project in the next path is resolved  
- Otherwise, keep the target project id&name
- Parse the project id/name from following paths without resolving it, and compare to the target (assuming they are all in the same project)